public/app/features/alerting/unified/hooks/useExternalAmSelector.ts

```diff
@@ -51,3 +58,71 @@ export function useExternalAmSelector(): AlertmanagerConfig[] | [] {
 
   return [...enabledAlertmanagers, ...droppedAlertmanagers];
 }
+
+export interface ExternalDataSourceAM {
+  dataSource: DataSourceInstanceSettings<AlertManagerDataSourceJsonData>;
+  url?: string;
+  status: 'active' | 'pending' | 'dropped';
+  statusInconclusive?: boolean;
+}
+
+export function useExternalDataSourceAlertmanagers(): ExternalDataSourceAM[] {
+  const externalDsAlertManagers = getAlertManagerDataSources().filter((ds) => ds.jsonData.handleGrafanaManagedAlerts);
+
+  const alertmanagerDatasources = useSelector((state: StoreState) =>
+    keyBy(
+      state.dataSources.dataSources.filter((ds) => ds.type === 'alertmanager'),
+      (ds) => ds.uid
+    )
+  );
+
+  const discoveredAlertmanagers = useUnifiedAlertingSelector(
+    (state) => state.externalAlertmanagers.discoveredAlertmanagers.result?.data
+  );
+
+  const droppedAMUrls = countBy(discoveredAlertmanagers?.droppedAlertManagers, (x) => x.url);
+  const activeAMUrls = countBy(discoveredAlertmanagers?.activeAlertManagers, (x) => x.url);
+
+  return externalDsAlertManagers.map<ExternalDataSourceAM>((dsAm) => {
+    const dsSettings = alertmanagerDatasources[dsAm.uid];
+
+    if (!dsSettings) {
+      return {
+        dataSource: dsAm,
+        status: 'pending',
+      };
+    }
+
+    const amUrl = getDataSourceUrlWithProtocol(dsSettings);
+    const amStatusUrl = `${amUrl}/api/v2/alerts`;
+
+    const matchingDroppedUrls = droppedAMUrls[amStatusUrl] ?? 0;
+    const matchingActiveUrls = activeAMUrls[amStatusUrl] ?? 0;
+
+    const isDropped = matchingDroppedUrls > 0;
+    const isActive = matchingActiveUrls > 0;
+
+    // It may happen that there are multiple Alertmanagers of the same URL (e.g. only different credentials)
+    // Alertmanager response contains only URLs so in case of duplication we are not able
+    // to distinguish which is which, hence the status is inconclusive
+    const isStatusInconclusive = matchingDroppedUrls + matchingActiveUrls > 1;
```

Comment:
```suggestion
    // Multiple Alertmanagers of the same URL may exist (e.g. with different credentials)
    // Alertmanager response only contains URLs, so in case of duplication, we are not able
    // to distinguish which is which, resulting in an inconclusive status. 
    const isStatusInconclusive = matchingDroppedUrls + matchingActiveUrls > 1;
```